pkg/controller/statefulset/stateful_set.go

```diff
@@ -153,16 +159,42 @@ func (ssc *StatefulSetController) Run(workers int, stopCh <-chan struct{}) {
 // addPod adds the statefulset for the pod to the sync queue
 func (ssc *StatefulSetController) addPod(obj interface{}) {
 	pod := obj.(*v1.Pod)
-	glog.V(4).Infof("Pod %s created, labels: %+v", pod.Name, pod.Labels)
-	set := ssc.getStatefulSetForPod(pod)
-	if set == nil {
+
+	if pod.DeletionTimestamp != nil {
+		// on a restart of the controller manager, it's possible a new pod shows up in a state that
+		// is already pending deletion. Prevent the pod from being a creation observation.
+		ssc.deletePod(pod)
```

Comment:
That check is done inside `deletePod()`. Is there a benefit to doing it before as well?